fido/exceptions.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+
+
+class BaseTimeoutError(Exception):
+    """
+    Base class for all errors due to timeouts.
+    """
+
+
+class ConnectTimeoutError(BaseTimeoutError):
+    """
+    Connection took too long to establish
+    """
+
+
+class HTTPTimeoutError(BaseTimeoutError):
+    """
```

Comment:
This all sounds good to me. I'm on the fence about the base class - it would help if you actually called out in the comments of the base class that its purpose is a no-nonsense catchall.